lua/lazyvim/plugins/extras/lang/python.lua

```diff
@@ -0,0 +1,116 @@
+return {
+  {
+    "nvim-treesitter/nvim-treesitter",
+    opts = function(_, opts)
+      if type(opts.ensure_installed) == "table" then
+        vim.list_extend(opts.ensure_installed, { "ninja", "python", "rst", "toml" })
+      end
+    end,
+  },
+  {
+    "jose-elias-alvarez/null-ls.nvim",
+    opts = function(_, opts)
+      local nls = require("null-ls")
+      opts.sources = opts.sources or {}
+      vim.list_extend(opts.sources, {
+        -- Order of formatters matters. They are used in order of appearance.
+        nls.builtins.formatting.ruff.with({ filetypes = { "python" }, prefer_local = ".venv/bin" }),
+        nls.builtins.formatting.black.with({ filetypes = { "python" }, prefer_local = ".venv/bin" }),
```

Comment:
I'm not sure why both of these are installed, imo ruff is far superior than black

Reply:
But ruff does not have all black features. Ruff is a replacement for isort.

Reply:
Ruff is used here instead od `isort`. I do not include diagnostics here from `ruff` because of `ruff_lsp`.

Reply:
ah, I thought ruff also had code formatting capabilities as well

Reply:
Would the entry of ruff here automatically apply any formatting other than the import ordering?

Reply:
> Would the entry of ruff here automatically apply any formatting other than the import ordering?

No. `ruff` can only organise imports exactly like `isort`. That's why black was also added to this configuration initially.